eventfiles/class_tests/event_file_views_tests.py

```diff
@@ -47,3 +61,46 @@ def test_unauthorized_user_redirected_to_main_page(self, client):
         response = client.get(response.url)
         assert response.status_code == 200
         assertTemplateUsed(response, 'main/index.html')
+
+    def test_delete_button_visible_for_file_user_created(self, sign_in_with_user2, client):
+        """ test that an ordinary user can delete only files he created """
+        response = client.get(reverse("event_files", args=[1]))
+        char_content = response.content.decode(response.charset)
+        found_file3 = re.search(f'delete testFile{FILE_CREATED_BY_USER2}', char_content)
+        found_file1 = re.search(f'delete testFile{FILE_CREATED_BY_EVENT_CREATOR}', char_content)
+        print(char_content)
+        assert found_file3 and not found_file1
+
+    def test_unauthorized_user_delete(self, sign_in_with_user2, client):
+        response = client.get(f'{FILE_DELETE_URL}/{FILE_CREATED_BY_EVENT_CREATOR}')
+        assert response.status_code == 401
+
+    def test_delete_file_created_by_user(self, sign_in_with_user2, client):
+        client.get(f'{FILE_DELETE_URL}/{FILE_CREATED_BY_USER2}')
+        assert not EventFile.objects.all().filter(id=FILE_CREATED_BY_USER2).exists()
+
+    def test_all_delete_buttons_visible_for_event_creator(self, sign_in_with_creator, client):
+        """ test that the event creator can delete all files """
+        response = client.get(reverse("event_files", args=[1]))
+        char_content = response.content.decode(response.charset)
+        found_file3 = re.search(f'delete testFile{FILE_CREATED_BY_USER2}', char_content)
+        found_file1 = re.search(f'delete testFile{FILE_CREATED_BY_EVENT_CREATOR}', char_content)
+        assert found_file3 and found_file1
+
+    def test_event_creator_delete_any_file(self, sign_in_with_creator, client):
+        client.get(f'{FILE_DELETE_URL}/{FILE_CREATED_BY_USER2}')
+        assert not EventFile.objects.all().filter(id=FILE_CREATED_BY_USER2).exists()
+
+    def test_unsigned_user_delete(self, client):
+        response = client.get(f'{FILE_DELETE_URL}/{FILE_CREATED_BY_USER2}')
```

Comment:
Can you create a constant for this statement: f'{FILE_DELETE_URL}/{FILE_CREATED_BY_USER2}'
It seems that you use it in almost every function so you can use a constant instead 😄

Reply:
Done :)